### ISSUE TITLE
Ubertooth-scan, Finding non-discoverable devices

### ISSUE THREAD

**Issue by garyv100** [greatscottgadgets/ubertooth#167]
Hi Dominic, 

I am trying to replicate your YouTube demo: HCI scan, Finding non-discoverable devices.

This is what I do and get:
/****************************************
sudo ./ubertooth-scan -s -x
HCI scan

Ubertooth scan

Scan results:
/****************************************
sudo ubertooth-scan -b hci0 -t 40 -s -x 
HCI scan

Ubertooth scan

Scan results:
/****************************************

I have just updated to your latest release as of 03/12/2016, rebuild and reloaded the firmware.
 ./ubertooth-util -V
ubertooth git- (garyv1@tech) Mon Mar 14 13:11:40 EDT 2016

./ubertooth-util -v
Firmware revision: git- 

ubertooth-rx -V
libubertooth 0.3 (GITDIR-NOTFOUND), libbtbb 0.4 (GITDIR-NOTFOUND)

I had the same issue w/ previous release: 2015-10-R1.

My environment: ubuntu 14.04 LTS on metal (no vm of any kind). I use Trandnet bluetooth along with Ubertooth One. I can successfully detect MAC of a bluetooth devices using: "hcitool inq".

Please correct me if i'm wrong, ubertooth-scan -s -x retrieves MAC of nearby bluetooth devices (i have a mix of Android and iPhone devices in one foot radius) in a non-discoverable mode or even if bluetooth is disabled - as long as they are powered? 

What am I doing wrong?

Best regards,

Gary

**Comment by dominicgs** [comment]
I'm not sure what you're doing wrong, it looks like your Ubertooth isn't seeing any data.

As you are running the git code, could you try using `ubertooth-rx -z` ?

**Comment by dominicgs** [comment]
Let me rephrase that - I don't think you're doing anything wrong, I think your Ubertooth is just not seeing any packets.

**Comment by garyv100** [comment]
Dominic, you are correct, the bt was not sending any data and "ubertooth-rx -z" returns no data.
If I, for example, transfer a file over bt - it does return data and I get the following:
/******************************
./ubertooth-scan -s -x
HCI scan

Ubertooth scan
systime=1458054711 ch= 2 LAP=3d6bb7 err=0 clk100ns=1303146539 clk1=1257079 s=-36 n=-74 snr=38
systime=1458054712 ch=66 LAP=3d6bb7 err=1 clk100ns=1303997799 clk1=1257215 s=-53 n=-73 snr=20
systime=1458054712 ch=19 LAP=3d6bb7 err=0 clk100ns=1304371648 clk1=1257275 s=-16 n=-67 snr=51

Scan results:
??:??:85:3D:6B:B7	LGMS323
Requesting information ...
Can't create connection: Operation not permitted
AFH map: 0x048f5808001034055c53
/*********************************

Please do correct me if i'm wrong. In order for Ubertooth One to retrieve MAC it must receive data packets from that device and in doing so it reconstructs MAC?  If it does not get data, it cannot do it?

Many thanks,

Gary

PS: Love your device, a lot of hard work and dedication went in to it.  In my previous life i used to do similar things - I can appreciate it.

**Comment by dominicgs** [comment]
That's correct, Ubertooth is sniffing packets and reconstructing the lower four bytes of the MAC.  You can then take those bytes and try to connect to the device (putting 00:00 for the first two bytes).

As you can see from the 'HCI scan' portion of your output, the device was not discoverable to the Bluetooth dongle.

**Comment by garyv100** [comment]
Dominic, thank you so much!

**Comment by bigoper** [comment]
Hi @garyv100 and @dominicgs 

I was hoping I could use your help here :)

# CURRENT STATE
### `ubertooth-scan -s -x`
- No BT device is being actively used used/sync/advertising
- I have laptops, phones, iPads near me with BT activated/on
```js
└─$ ubertooth-scan -s -x
HCI scan

Ubertooth scan
systime=1698884138 ch=26 LAP=07e88f err=2 clk100ns=26085068 clk1=4174 s=-80 n=-55 snr=-25
systime=1698884146 ch=66 LAP=a383e7 err=2 clk100ns=97813755 clk1=15650 s=-81 n=-55 snr=-26

Scan results:
??:??:??:07:E8:8F
AFH map: 0x00000004000000000000
??:??:??:A3:83:E7
AFH map: 0x00000000000000000400
```
```js
└─$ ubertooth-scan -s -x
HCI scan

Ubertooth scan
systime=1698884387 ch=72 LAP=a9dfd5 err=1 clk100ns=40549063 clk1=6488 s=-75 n=-55 snr=-20

Scan results:
??:??:??:A9:DF:D5
AFH map: 0x00000000000000000001
```
- You can see that the results are NOT consistent.

### `ubertooth-rx -z`
- Same devices as before, same state.
```js
└─$ ubertooth-rx -z     
systime=1698884436 ch=74 LAP=a9dfd5 err=0 clkn=33837 clk_offset=4491 s=-74 n=-55 snr=-19
systime=1698884437 ch=67 LAP=a9dfd5 err=1 clkn=36653 clk_offset=4354 s=-76 n=-55 snr=-21
systime=1698884440 ch=69 LAP=a9dfd5 err=2 clkn=47397 clk_offset=4160 s=-74 n=-55 snr=-19
systime=1698884451 ch=68 LAP=a9dfd5 err=2 clkn=82533 clk_offset=3654 s=-75 n=-55 snr=-20
systime=1698884460 ch= 2 LAP=a9dfd5 err=2 clkn=111913 clk_offset=3454 s=-73 n=-55 snr=-18
systime=1698884463 ch=66 LAP=a9dfd5 err=2 clkn=122313 clk_offset=3281 s=-74 n=-55 snr=-19
systime=1698884479 ch= 4 LAP=0c3711 err=1 clkn=173297 clk_offset=4127 s=-77 n=-55 snr=-22
systime=1698884491 ch=77 LAP=a9dfd5 err=2 clkn=211712 clk_offset=2256 s=-72 n=-55 snr=-17
systime=1698884524 ch=72 LAP=a9dfd5 err=1 clkn=316376 clk_offset=909 s=-78 n=-55 snr=-23
systime=1698884540 ch=70 LAP=465b37 err=0 clkn=366254 clk_offset=2489 s=-63 n=-55 snr=-8
systime=1698884540 ch=70 LAP=465b37 err=0 clkn=366256 clk_offset=2474 s=-54 n=-55 snr=1
systime=1698884541 ch=69 LAP=a9dfd5 err=1 clkn=371028 clk_offset=125 s=-75 n=-55 snr=-20
systime=1698884561 ch=60 LAP=a9dfd5 err=0 clkn=433799 clk_offset=5218 s=-74 n=-55 snr=-19
systime=1698884571 ch=66 LAP=a9dfd5 err=1 clkn=466155 clk_offset=4829 s=-74 n=-55 snr=-19
systime=1698884586 ch=77 LAP=a9dfd5 err=1 clkn=515107 clk_offset=4120 s=-73 n=-55 snr=-18
systime=1698884619 ch= 8 LAP=c360de err=2 clkn=619480 clk_offset=2767 s=-82 n=-55 snr=-27
systime=1698884620 ch=50 LAP=07eae6 err=2 clkn=622821 clk_offset=4551 s=-75 n=-55 snr=-20
systime=1698884652 ch=60 LAP=07eae6 err=2 clkn=727132 clk_offset=314 s=-78 n=-55 snr=-23
systime=1698884656 ch=68 LAP=a9dfd5 err=2 clkn=739818 clk_offset=956 s=-74 n=-55 snr=-19
^CSurvey Results
??:??:??:A9:DF:D5
??:??:??:0C:37:11
??:??:??:46:5B:37
??:??:??:C3:60:DE
??:??:??:07:EA:E6
```
- I'm detecting more devices.

# NEW STATE
- Same devices, but now they'll be in discoverable mode.
- I'm doing this^ so I can compare the results to make sure that LAP that I'm seeing are my devices.

```js
└─$ ubertooth-rx -z     
systime=1698885286 ch=43 LAP=4201ad err=2 clkn=140183 clk_offset=4901 s=-61 n=-55 snr=-6
systime=1698885289 ch=69 LAP=484d07 err=2 clkn=148577 clk_offset=4489 s=-72 n=-55 snr=-17
systime=1698885294 ch=27 LAP=9e8b33 err=0 clkn=165401 clk_offset=4401 s=-49 n=-55 snr=6
systime=1698885294 ch=27 LAP=9e8b33 err=0 clkn=165433 clk_offset=4399 s=-51 n=-55 snr=4
systime=1698885294 ch=27 LAP=9e8b33 err=0 clkn=165465 clk_offset=4418 s=-50 n=-55 snr=5
systime=1698885294 ch=27 LAP=9e8b33 err=0 clkn=165497 clk_offset=4412 s=-49 n=-55 snr=6
systime=1698885294 ch=59 LAP=9e8b33 err=1 clkn=165590 clk_offset=1278 s=-54 n=-55 snr=1
systime=1698885294 ch=29 LAP=9e8b33 err=2 clkn=166049 clk_offset=4395 s=-48 n=-55 snr=7
systime=1698885294 ch=29 LAP=9e8b33 err=2 clkn=166081 clk_offset=4390 s=-49 n=-55 snr=6
systime=1698885294 ch=61 LAP=9e8b33 err=0 clkn=166206 clk_offset=1263 s=-56 n=-55 snr=-1
systime=1698885294 ch=61 LAP=9e8b33 err=0 clkn=166238 clk_offset=1278 s=-56 n=-55 snr=-1
systime=1698885294 ch=31 LAP=9e8b33 err=0 clkn=166665 clk_offset=4372 s=-48 n=-55 snr=7
systime=1698885294 ch=31 LAP=9e8b33 err=0 clkn=166729 clk_offset=4369 s=-49 n=-55 snr=6
systime=1698885294 ch=31 LAP=9e8b33 err=1 clkn=166761 clk_offset=4375 s=-49 n=-55 snr=6
systime=1698885295 ch=33 LAP=9e8b33 err=0 clkn=167377 clk_offset=4353 s=-50 n=-55 snr=5
systime=1698885295 ch=65 LAP=9e8b33 err=0 clkn=167534 clk_offset=1220 s=-61 n=-55 snr=-6
systime=1698885296 ch=45 LAP=9e8b33 err=2 clkn=171197 clk_offset=4264 s=-50 n=-55 snr=5
systime=1698885296 ch=45 LAP=9e8b33 err=0 clkn=171261 clk_offset=4258 s=-50 n=-55 snr=5
systime=1698885296 ch=77 LAP=9e8b33 err=0 clkn=171290 clk_offset=1126 s=-65 n=-55 snr=-10
systime=1698885296 ch=77 LAP=9e8b33 err=2 clkn=171322 clk_offset=1129 s=-67 n=-55 snr=-12
systime=1698885296 ch=77 LAP=9e8b33 err=0 clkn=171354 clk_offset=1128 s=-66 n=-55 snr=-11
systime=1698885296 ch=51 LAP=9e8b33 err=2 clkn=173077 clk_offset=4213 s=-54 n=-55 snr=1
systime=1698885296 ch=51 LAP=9e8b33 err=0 clkn=173173 clk_offset=4215 s=-54 n=-55 snr=1
systime=1698885297 ch=53 LAP=9e8b33 err=0 clkn=173725 clk_offset=4190 s=-53 n=-55 snr=2
systime=1698885297 ch=53 LAP=9e8b33 err=0 clkn=173757 clk_offset=4204 s=-53 n=-55 snr=2
systime=1698885297 ch=53 LAP=9e8b33 err=1 clkn=173789 clk_offset=4201 s=-53 n=-55 snr=2
systime=1698885297 ch= 6 LAP=9e8b33 err=2 clkn=173882 clk_offset=1065 s=-47 n=-55 snr=8
systime=1698885297 ch= 6 LAP=9e8b33 err=1 clkn=173914 clk_offset=1058 s=-44 n=-55 snr=11
systime=1698885297 ch= 6 LAP=9e8b33 err=2 clkn=173946 clk_offset=1058 s=-47 n=-55 snr=8
systime=1698885297 ch=55 LAP=9e8b33 err=0 clkn=174341 clk_offset=4183 s=-53 n=-55 snr=2
systime=1698885297 ch=55 LAP=9e8b33 err=1 clkn=174373 clk_offset=4163 s=-53 n=-55 snr=2
systime=1698885297 ch=55 LAP=9e8b33 err=1 clkn=174437 clk_offset=4181 s=-53 n=-55 snr=2
systime=1698885297 ch= 8 LAP=9e8b33 err=0 clkn=174466 clk_offset=1049 s=-46 n=-55 snr=9
systime=1698885297 ch= 8 LAP=9e8b33 err=0 clkn=174498 clk_offset=1041 s=-47 n=-55 snr=8
systime=1698885297 ch= 8 LAP=9e8b33 err=1 clkn=174530 clk_offset=1044 s=-45 n=-55 snr=10
systime=1698885297 ch= 8 LAP=9e8b33 err=1 clkn=174562 clk_offset=1041 s=-48 n=-55 snr=7
systime=1698885297 ch=57 LAP=9e8b33 err=0 clkn=175085 clk_offset=4161 s=-53 n=-55 snr=2
systime=1698885297 ch=10 LAP=9e8b33 err=0 clkn=175146 clk_offset=1041 s=-47 n=-55 snr=8
systime=1698885297 ch=10 LAP=9e8b33 err=0 clkn=175178 clk_offset=1024 s=-48 n=-55 snr=7
^CSurvey Results
??:??:??:42:01:AD
??:??:??:48:4D:07
??:??:??:9E:8B:33
```

# THE ISSUE/QUESTION(S)
- It seems like all Apple devices are being captured as `??:??:??:9E:8B:33`, not sure why?
This make the detection of different Apple devices very difficult

### Turning on my BT headphones
> these^ are connected to two devices at the same time
```js
└─$ ubertooth-rx -z
systime=1698885612 ch=52 LAP=5e749b err=0 clkn=47028 clk_offset=1105 s=-50 n=-55 snr=5
systime=1698885612 ch=37 LAP=465b37 err=0 clkn=47314 clk_offset=2658 s=-60 n=-55 snr=-5
systime=1698885612 ch=69 LAP=465b37 err=2 clkn=47430 clk_offset=2669 s=-63 n=-55 snr=-8
systime=1698885612 ch=54 LAP=5e749b err=0 clkn=47676 clk_offset=1081 s=-50 n=-55 snr=5
systime=1698885612 ch=54 LAP=5e749b err=0 clkn=47740 clk_offset=1078 s=-51 n=-55 snr=4
systime=1698885612 ch=71 LAP=5e749b err=2 clkn=48073 clk_offset=4184 s=-55 n=-55 snr=0
systime=1698885612 ch=71 LAP=5e749b err=0 clkn=48105 clk_offset=4187 s=-56 n=-55 snr=-1
systime=1698885612 ch=73 LAP=5e749b err=0 clkn=48721 clk_offset=4166 s=-54 n=-55 snr=1
systime=1698885612 ch=73 LAP=5e749b err=2 clkn=48753 clk_offset=4171 s=-55 n=-55 snr=0
systime=1698885613 ch=77 LAP=465b37 err=0 clkn=50022 clk_offset=2663 s=-63 n=-55 snr=-8
systime=1698885613 ch=47 LAP=5e749b err=1 clkn=50480 clk_offset=955 s=-47 n=-55 snr=8
systime=1698885613 ch=47 LAP=465b37 err=0 clkn=50514 clk_offset=2621 s=-58 n=-55 snr=-3
systime=1698885613 ch=32 LAP=5e749b err=0 clkn=50736 clk_offset=963 s=-45 n=-55 snr=10
systime=1698885613 ch=17 LAP=5e749b err=2 clkn=51048 clk_offset=956 s=-53 n=-55 snr=2
systime=1698885613 ch=36 LAP=5e749b err=0 clkn=52052 clk_offset=912 s=-47 n=-55 snr=8
systime=1698885613 ch=68 LAP=5e749b err=0 clkn=52182 clk_offset=913 s=-51 n=-55 snr=4
systime=1698885614 ch=40 LAP=5e749b err=0 clkn=53354 clk_offset=897 s=-45 n=-55 snr=10
systime=1698885614 ch=57 LAP=5e749b err=0 clkn=53704 clk_offset=869 s=-49 n=-55 snr=6
systime=1698885614 ch=10 LAP=5e749b err=0 clkn=53832 clk_offset=866 s=-49 n=-55 snr=6
systime=1698885614 ch=10 LAP=5e749b err=1 clkn=53834 clk_offset=878 s=-49 n=-55 snr=6
systime=1698885614 ch=44 LAP=465b37 err=0 clkn=54530 clk_offset=2564 s=-57 n=-55 snr=-2
systime=1698885614 ch=44 LAP=5e749b err=0 clkn=54576 clk_offset=845 s=-44 n=-55 snr=11
systime=1698885614 ch=46 LAP=465b37 err=1 clkn=55214 clk_offset=2606 s=-57 n=-55 snr=-2
systime=1698885614 ch=46 LAP=465b37 err=1 clkn=55216 clk_offset=2625 s=-47 n=-55 snr=8
systime=1698885615 ch=48 LAP=465b37 err=0 clkn=55890 clk_offset=2602 s=-57 n=-55 snr=-2
systime=1698885615 ch=48 LAP=465b37 err=0 clkn=55892 clk_offset=2598 s=-48 n=-55 snr=7
systime=1698885615 ch=65 LAP=465b37 err=0 clkn=56270 clk_offset=2567 s=-60 n=-55 snr=-5
systime=1698885615 ch=20 LAP=5e749b err=0 clkn=57068 clk_offset=849 s=-54 n=-55 snr=1
UAP = 0x88 found after 14 total packets.
systime=1698885615 ch=73 LAP=465b37 err=1 clkn=58834 clk_offset=2582 s=-66 n=-55 snr=-11
systime=1698885616 ch=43 LAP=465b37 err=0 clkn=59310 clk_offset=2556 s=-58 n=-55 snr=-3
systime=1698885616 ch=30 LAP=5e749b err=0 clkn=60214 clk_offset=883 s=-47 n=-55 snr=8
systime=1698885616 ch=32 LAP=465b37 err=0 clkn=60852 clk_offset=2513 s=-45 n=-55 snr=10
systime=1698885617 ch= 6 LAP=465b37 err=0 clkn=62638 clk_offset=2474 s=-52 n=-55 snr=3
systime=1698885617 ch=38 LAP=465b37 err=0 clkn=62766 clk_offset=2467 s=-60 n=-55 snr=-5
systime=1698885617 ch=38 LAP=465b37 err=0 clkn=62802 clk_offset=2462 s=-60 n=-55 snr=-5
systime=1698885617 ch=70 LAP=465b37 err=0 clkn=62898 clk_offset=2458 s=-58 n=-55 snr=-3
systime=1698885617 ch=55 LAP=465b37 err=0 clkn=63202 clk_offset=2450 s=-52 n=-55 snr=3
systime=1698885617 ch= 8 LAP=465b37 err=0 clkn=63310 clk_offset=2447 s=-53 n=-55 snr=2
systime=1698885617 ch=74 LAP=465b37 err=0 clkn=64198 clk_offset=2414 s=-61 n=-55 snr=-6
systime=1698885617 ch=27 LAP=465b37 err=0 clkn=64358 clk_offset=2391 s=-64 n=-55 snr=-9
systime=1698885617 ch=29 LAP=5e749b err=0 clkn=64944 clk_offset=868 s=-45 n=-55 snr=10
systime=1698885618 ch=65 LAP=465b37 err=2 clkn=66350 clk_offset=2347 s=-60 n=-55 snr=-5
systime=1698885618 ch=69 LAP=465b37 err=0 clkn=67698 clk_offset=2361 s=-60 n=-55 snr=-5
systime=1698885618 ch=69 LAP=465b37 err=0 clkn=67700 clk_offset=2353 s=-60 n=-55 snr=-5
systime=1698885618 ch=39 LAP=465b37 err=0 clkn=68178 clk_offset=2335 s=-60 n=-55 snr=-5
systime=1698885618 ch=71 LAP=5e749b err=1 clkn=68272 clk_offset=836 s=-52 n=-55 snr=3
systime=1698885619 ch=60 LAP=465b37 err=0 clkn=69798 clk_offset=2292 s=-57 n=-55 snr=-2
systime=1698885619 ch=60 LAP=465b37 err=1 clkn=69862 clk_offset=2291 s=-56 n=-55 snr=-1
systime=1698885619 ch=77 LAP=465b37 err=0 clkn=70248 clk_offset=2272 s=-45 n=-55 snr=10
systime=1698885619 ch=47 LAP=465b37 err=1 clkn=70702 clk_offset=2270 s=-57 n=-55 snr=-2
systime=1698885619 ch=47 LAP=465b37 err=0 clkn=70758 clk_offset=2285 s=-57 n=-55 snr=-2
systime=1698885619 ch= 0 LAP=5e749b err=0 clkn=70816 clk_offset=806 s=-51 n=-55 snr=4
systime=1698885619 ch=64 LAP=465b37 err=0 clkn=71126 clk_offset=2262 s=-54 n=-55 snr=1
systime=1698885619 ch=64 LAP=465b37 err=0 clkn=71128 clk_offset=2251 s=-54 n=-55 snr=1
systime=1698885619 ch=49 LAP=465b37 err=2 clkn=71366 clk_offset=2262 s=-54 n=-55 snr=1
systime=1698885619 ch=34 LAP=465b37 err=0 clkn=71630 clk_offset=2240 s=-58 n=-55 snr=-3
systime=1698885619 ch=34 LAP=465b37 err=0 clkn=71632 clk_offset=2245 s=-53 n=-55 snr=2
systime=1698885620 ch=68 LAP=465b37 err=0 clkn=72392 clk_offset=2234 s=-56 n=-55 snr=-1
systime=1698885620 ch=55 LAP=465b37 err=0 clkn=73256 clk_offset=2234 s=-53 n=-55 snr=2
systime=1698885620 ch=40 LAP=465b37 err=0 clkn=73582 clk_offset=2223 s=-54 n=-55 snr=1
systime=1698885620 ch=40 LAP=465b37 err=0 clkn=73584 clk_offset=2218 s=-54 n=-55 snr=1
systime=1698885620 ch=72 LAP=240d6e err=2 clkn=73726 clk_offset=276 s=-87 n=-55 snr=-32
systime=1698885621 ch=65 LAP=465b37 err=0 clkn=76498 clk_offset=2238 s=-53 n=-55 snr=2
systime=1698885621 ch=52 LAP=465b37 err=0 clkn=77390 clk_offset=2207 s=-53 n=-55 snr=2
systime=1698885621 ch=52 LAP=465b37 err=1 clkn=77392 clk_offset=2198 s=-53 n=-55 snr=2
^CSurvey Results
??:??:??:35:CC:1B
??:??:4F:46:5B:37
??:??:C2:59:A7:C4
??:??:88:5E:74:9B
??:??:??:24:0D:6E
```
- My Apple devices are **not** in the list of devices found.
Not too sure how this thing works/operates, as the results are confusing.

# KISMET / BLUE HYDRA
- Have you guys tried any of these? and was also successful?

Thanks for looking!
